DataFormats/Detectors/FIT/FT0/include/DataFormatsFT0/RawEventData.h

```diff
@@ -0,0 +1,149 @@
+// Copyright CERN and copyright holders of ALICE O2. This software is
+// distributed under the terms of the GNU General Public License v3 (GPL
+// Version 3), copied verbatim in the file "COPYING".
+//
+// See http://alice-o2.web.cern.ch/license for full licensing information.
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+//
+//file RawEventData.h class  for RAW data format
+//Alla.Maevskaya@cern.ch
+// with Artur.Furs
+//
+#ifndef ALICEO2_FIT_RAWEVENTDATA_H_
+#define ALICEO2_FIT_RAWEVENTDATA_H_
+
+#include "DataFormatsFT0/Digit.h"
+#include "Headers/RAWDataHeader.h"
+#include "DataFormatsFT0/LookUpTable.h"
+#include <CommonDataFormat/InteractionRecord.h>
+#include <iostream>
+#include <bitset>
+#include "Rtypes.h"
+namespace o2
+{
+namespace ft0
+{
+constexpr int NCHANNELS_FT0 = 208;
+constexpr int NCHANNELS_PM = 12;
+constexpr int NPMs = 18;
+constexpr int NBITS_EVENTDATA = 9;
+
+struct EventHeader {
+  ushort startDescriptor : 4;
+  uint reservedField : 28;
+  uint orbit : 32;
+  short bc : 12;
+  short nGBTWords : 4;
+  ClassDefNV(EventHeader, 1);
+};
+struct EventData {
+  short int time : 12;
+  short int charge : 12;
+  unsigned short int numberADC : 1;
+  bool isDoubleEvent : 1;
+  bool is1TimeLostEvent : 1;
+  bool is2TimeLostEvent : 1;
+  bool isADCinGate : 1;
+  bool isTimeInfoLate : 1;
+  bool isAmpHigh : 1;
+  bool isEventInTVDC : 1;
+  bool isTimeInfoLost : 1;
+  uint channelID : 4;
+  ClassDefNV(EventData, 1);
+};
```

Comment:
For the same reason as for EventHeader, please use
````
struct EventData {
  static constexpr int PayloadSize = 5; // size in bytes = 1/2 GBT word
  union {
    uint64_t w = 0;
    struct {
      uint16_t time : 12;
      uint16_t charge : 12;
      uint8_t numberADC : 1;
      uint8_t isDoubleEvent : 1;
      uint8_t is1TimeLostEvent : 1;
      uint8_t is2TimeLostEvent : 1;
      uint8_t isADCinGate : 1;
      uint8_t isTimeInfoLate : 1;
      uint8_t isAmpHigh : 1;
      uint8_t isEventInTVDC : 1;
      uint8_t isTimeInfoLost : 1;
      uint8_t channelID : 4;
    };
  };
  ClassDefNV(EventData, 1);
};
````

Reply:
time and charge are signed : time is placed around 0 ( 0 is middle of the bunch) and charge could be negative as well

Reply:
ok, then change them to int16_t

Reply:
why do you want `uint8_t`? This broke my printout. May be  just `unsigned` ?

Reply:
to have portable types, but in this case you can use unsigned or bool also.